stingray/lightcurve.py

```diff
@@ -526,23 +524,24 @@ def make_lightcurve(toa, dt, tseg=None, tstart=None, gti=None, mjdref=0):
         # are not throwing away good events.
 
         if tseg is None:
-            tseg = toa[-1] - toa[0]
+            tseg = toa[-1] - tstart
 
         logging.info("make_lightcurve: tseg: " + str(tseg))
 
-        timebin = np.int(tseg/dt)
+        timebin = np.int64(tseg/dt)
         logging.info("make_lightcurve: timebin:  " + str(timebin))
 
-        tend = tstart + timebin*dt
-
-        counts, histbins = np.histogram(toa, bins=timebin,
-                                        range=[tstart, tend])
-
-        dt = histbins[1] - histbins[0]
-
-        time = histbins[:-1] + 0.5*dt
-
-        counts = np.asarray(counts)
+        tend = tstart + timebin * dt
+        good = (tstart <= toa) & (toa < tend)
+        if not use_hist:
+            binned_toas = ((toa[good] - tstart) // dt).astype(np.int64)
+            counts = \
+                np.bincount(binned_toas, minlength=timebin)
+            time = tstart + np.arange(0.5, 0.5 + len(counts)) * dt
+        else:
+            counts, histbins = np.histogram(toa[good],
+                                            bins=np.linspace(tstart, tend, 1 + timebin))
```

Comment:
... and here's a use of `np.linspace` we might want to replace with `np.arange` for speed?

Reply:
Done. Hope there are no side effects when the observation is very long?